xapian-bindings/java/SmokeTest.java

```diff
@@ -121,8 +121,19 @@ public static void main(String[] args) throws Exception {
 		System.exit(1);
 	    }
 	    Enquire enq = new Enquire(db);
-	    enq.setQuery(new Query(Query.OP_OR, "there", "is"));
+	    enq.setQuery(new Query("dog"));
 	    MSet mset = enq.getMSet(0, 10);
+	    // MSet size is 0
+	    if(mset.size() != 0) {
+	    	for(MSetIterator.MatchDescriptor descriptor : mset) {
+	    		System.err.println("Empty iterator entered invalid section (from MSet java iterator)");
+	    		break;
+	    	}
+	    }else {
+	    	System.err.println("Unexpected mset.size()");
+	    }
```

Comment:
You could also (as well as the above) explicitly test the `hasNext()` and `next()` behaviour, so you're not just testing the full iterator integration. (It'll make it slightly more obvious if one part of this fails in future.)